common/poller/FlatMappedPoller.java

```diff
@@ -18,36 +18,43 @@
 
 package com.vaticle.typedb.core.common.poller;
 
-import com.vaticle.typedb.core.common.iterator.FunctionalIterator;
-import com.vaticle.typedb.core.common.iterator.Iterators;
-
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Optional;
 import java.util.function.Function;
 
 public class FlatMappedPoller<T, U> extends AbstractPoller<U> {
 
     private final Poller<T> source;
-    private final Function<T, FunctionalIterator<U>> flatMappingFn;
-    private FunctionalIterator<U> currentIterator;
+    private final Function<T, Poller<U>> flatMappingFn;
+    private final List<Poller<U>> flatMappingPollers;
```

Comment:
i believe these are the `flatMappedPollers` right rather than `Mapping` pollers, they are already the outcome of the map operation?